store/generated_key/eigenda_v2/eigenda.go

```diff
@@ -0,0 +1,146 @@
+package eigendav2
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/avast/retry-go/v4"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/Layr-Labs/eigenda/api/clients/v2"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type Config struct {
+	// address of service manager - used for loading chain state
+	ServiceManagerAddr string
+
+	MaxBlobSizeBytes uint64
+
+	// total duration time that client waits for blob to confirm
+	StatusQueryTimeout time.Duration
+
+	// number of times to retry eigenda blob dispersals
+	PutRetries uint
+}
+
+// Store does storage interactions and verifications for blobs with the
+// EigenDA V2 protocol.
+type Store struct {
+	cfg *Config
+	log log.Logger
+
+	disperser *clients.PayloadDisperser
+	retriever clients.PayloadRetriever
+	verifier  verification.ICertVerifier
+}
+
+var _ common.GeneratedKeyStore = (*Store)(nil)
+
+func NewStore(log log.Logger, cfg *Config,
+	disperser *clients.PayloadDisperser, retriever clients.PayloadRetriever, verifier verification.ICertVerifier) (*Store, error) {
+	return &Store{
+		log:       log,
+		cfg:       cfg,
+		disperser: disperser,
+		retriever: retriever,
+		verifier:  verifier,
+	}, nil
+}
+
+// Get fetches a blob from DA using certificate fields and verifies blob
+// against commitment to ensure data is valid and non-tampered.
+func (e Store) Get(ctx context.Context, key []byte) ([]byte, error) {
+	var cert verification.EigenDACert
+	err := rlp.DecodeBytes(key, &cert)
+	if err != nil {
+		return nil, fmt.Errorf("RLP decoding EigenDA v2 cert: %w", err)
+	}
+
+	payload, err := e.retriever.GetPayload(ctx, &cert)
+	if err != nil {
+		return nil, fmt.Errorf("getting payload: %w", err)
+	}
+
+	return payload, nil
+}
+
+// Put disperses a blob for some pre-image and returns the associated RLP encoded certificate commit.
+// TODO: Client polling for different status codes
+//
+//	Mapping status codes to 503 failover
+func (e Store) Put(ctx context.Context, value []byte) ([]byte, error) {
+	salt := uint32(0)
+	e.log.Debug("Dispersing payload for EigenDA V2 network")
+
+	// TODO: Verify this retry or failover code for correctness against V2
+	// protocol
+
+	// We attempt to disperse the blob to EigenDA up to 3 times, unless we get a 400 error on any attempt.
+	cert, err := retry.DoWithData(
+		func() (*verification.EigenDACert, error) {
+			// TODO: Figure out salt mgmt
+			return e.disperser.SendPayload(ctx, value, salt)
+		},
+		retry.RetryIf(func(err error) bool {
+			salt++ // increment salt before retrying
```

Comment:
Should we actually increment salt on failures? My limited understanding was that we'd only need to increment salt if there is a _success_, and then a reorg happens that requires blob resubmission